Fix plex tests broken by Python 3.14.3 asyncio changes

### Description

The following `plex` tests are flaky/failing due to the Python 3.14.3 asyncio executor scheduling change (see home-assistant/core#162263 for background).

CPython PR python/cpython#142358 changed how `run_in_executor` delivers results back to the event loop. Results are now set directly on the asyncio future instead of going through `call_soon_threadsafe`, which means `await` returns immediately but other `call_soon_threadsafe` callbacks may not have run yet.

#### Affected tests

**`tests/components/plex/test_sensor.py::test_library_sensor_values` (failed 6/10)**

```
tests/components/plex/test_sensor.py:144: in test_library_sensor_values
    assert (
E   AssertionError: assert 'TV Show' == 'TV Show - S01E05 - Episode 5'
E     
E     - TV Show - S01E05 - Episode 5
E     + TV Show
```


#### What was done

These tests have been temporarily marked with `@pytest.mark.xfail` in PR #169069.

#### What needs to happen

The codeowner should investigate and fix the underlying timing issue in these tests. Common fixes include:
- Adding `await hass.async_block_till_done()` after service calls
- Converting sync entity methods to async
- Restructuring mocks to avoid unintended executor code paths

### Additional context

- Related PR: home-assistant/core#162263
- Disable PR: #169069

--- a/tests/components/plex/test_sensor.py
+++ b/tests/components/plex/test_sensor.py
@@ -72,9 +72,6 @@ class MockPlexTVEpisode(MockPlexMedia):
     parentYear = 2021
 
 
-@pytest.mark.xfail(
-    reason="Flaky due to Python 3.14.3 asyncio changes - see home-assistant/core#162263"
-)
 async def test_library_sensor_values(
     hass: HomeAssistant,
     entity_registry: er.EntityRegistry,
@@ -127,17 +124,16 @@ async def test_library_sensor_values(
     )
     await hass.async_block_till_done()
 
-    async_fire_time_changed(
-        hass,
-        dt_util.utcnow() + timedelta(seconds=RELOAD_AFTER_UPDATE_DELAY + 1),
-    )
-
     media = [MockPlexTVEpisode()]
     with patch(
         "plexapi.library.LibrarySection.recentlyAdded",
         return_value=media,
         __qualname__="recentlyAdded",
     ):
+        async_fire_time_changed(
+            hass,
+            dt_util.utcnow() + timedelta(seconds=RELOAD_AFTER_UPDATE_DELAY + 1),
+        )
         await hass.async_block_till_done()
 
     library_tv_sensor = hass.states.get("sensor.plex_server_1_library_tv_shows")
@@ -206,17 +202,16 @@ async def test_library_sensor_values(
     )
     await hass.async_block_till_done()
 
-    async_fire_time_changed(
-        hass,
-        dt_util.utcnow() + timedelta(seconds=RELOAD_AFTER_UPDATE_DELAY + 1),
-    )
-
     media = [MockPlexMovie()]
     with patch(
         "plexapi.library.LibrarySection.recentlyAdded",
         return_value=media,
         __qualname__="recentlyAdded",
     ):
+        async_fire_time_changed(
+            hass,
+            dt_util.utcnow() + timedelta(seconds=RELOAD_AFTER_UPDATE_DELAY + 1),
+        )
         await hass.async_block_till_done()
 
     library_movies_sensor = hass.states.get("sensor.plex_server_1_library_movies")
@@ -250,17 +245,16 @@ async def test_library_sensor_values(
     )
     await hass.async_block_till_done()
 
-    async_fire_time_changed(
-        hass,
-        dt_util.utcnow() + timedelta(seconds=RELOAD_AFTER_UPDATE_DELAY + 1),
-    )
-
     media = [MockPlexMusic()]
     with patch(
         "plexapi.library.LibrarySection.recentlyAdded",
         return_value=media,
         __qualname__="recentlyAdded",
     ):
+        async_fire_time_changed(
+            hass,
+            dt_util.utcnow() + timedelta(seconds=RELOAD_AFTER_UPDATE_DELAY + 1),
+        )
         await hass.async_block_till_done()
 
     library_music_sensor = hass.states.get("sensor.plex_server_1_library_music")

